### DISCUSSION TITLE
Release 6.0.0

### DISCUSSION THREAD

**Discussion by jgromes** [jgromes/RadioLib#737] [General]
# Breaking change

This release introduces the following backwards-incompatible changes to the public API:

* All methods to attach interrupts (e.g., `setDioAction()`, `setGdoAction()` etc.) no longer have a default level change direction.
* Signature of all methods/functions that use pin numbers (e.g. the `Module` constructor) changed from platform-dependent "pin type" changed to a common type (uint32_t)
* Reworked hardware abstraction

Other changes:

* [MOD] Fixed some signed comparison warnings
* [SX126x] Added missing XTAL check
* Added further explanation of the -707 error code (#691)
* [SX126x] Fixed calibration order (#689)
* [SX126x] Added spectral scan
* [SX126x] Added patch binary license
* [SX127x] Added missing standby for PhysicalLayer (#695)
* [nRF24] Added missing standby overload
* [RF69] Added missing standby overload
* [Si443x] Added missing standby overload
* [CC1101] Added missing standby overload
* [Pager] Added missing exclude direct receive guards (#697)
* add functions to PhysicalLayer interface (by @gasagna)
* [SX126x] Added frequency error reading
* [SX126x] Fixed chip id (#707)
* [SX126x] Fix failure to enable low power sleep on CubeCell AB01 (by @gasagna)
* [SX126x] Fixed frequency error calculation (#706)
* [SX127x] Explicitly enabled CRC (#706)
* [SX126x] Explicitly enabled CRC (#706)
* [SX1278] New getInstRSSI to get current RSSI
* [PHY] Fixed startReceive compatibility (#700)
* [SX126x] Added spectral scan in frequency
* [SX127x] & [SX126x] read current RSSI for getRSSI
* [SX126x] Unified getRSSI interface
* [SX127x] Unified getRSSI interface
* [SX126x] Fix intermittent failure to initialise cubecell (#706) (by @gasagna)
* [MOD] Fixed swapped NSS/BUSY order (#716)
* [SX126x] Fixed packet length in LoRa implicit mode
* [SX128x] Fixed packet length in implicit mode (#716)
* [SX126x] Improved RNG
* [SX127x] Fixed comments
* [SX126x] Added IQ inversion
* [SX128x] Added IQ inversion (#724)
* [SX127x] Implemented fractional bit rate (#505)
* [STM32WL] Fixed build for Platformio (#718)
* [SX127x] Added missing explicit CRC configuration
* [SX127x] Removed standby before data read
* [SX126x] Remove standby before data read (#703)
* [SX126x] Fixed IQ inversion not caching (#731)
* [SX128x] Fixed IQ inversion
* [HAL] Improve hardware abstraction layer (#730) (by @Mesteery)
* [Pager] Fixed deprecation warning for ESP
* [*] General reformatting
* [MOD] Added debug message when BUSY pin times out
* [MOD] Decreased default SPI timeout to 1000 ms

<hr /><em>This discussion was created from the release <a href='https://github.com/jgromes/RadioLib/releases/tag/6.0.0'>Release 6.0.0</a>.</em>

**Comment by timmbogner** [comment]
Hi Jan,
>All methods to attach interrupts no longer have a default level change direction.

This seems to only apply to SX127x currently, and not SX126x (or 8x). Do you plan to extend the new rule to those as well?

**Reply by jgromes** [reply]
This was done on CC1101 and SX127x because those already had a default value for the parameter direction. Because of the changes to HAL, it was no longer possible to have that. SX126x and others currently do not expose the direction parameter at all, so there was no change to the public API there.

**Comment by bjfhskj** [comment]
With module LLCC68, when I use 6.0.0 or 5.7.0，I get error : 'RADIOLIB_ERR_CHIP_NOT_FOUND, but 5.6.0 all is right,

**Reply by jgromes** [reply]
Please open a new issue and provide all of the information asked for by the issue template, especially your code, hardware setup and debug + verbose output.
